control/control_grpc_server_test.go

```diff
@@ -0,0 +1,699 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package control
+
+import (
+	"bytes"
+	"encoding/gob"
+	"io/ioutil"
+	"net"
+	"path"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/intelsdi-x/snap/control/plugin"
+	"github.com/intelsdi-x/snap/control/rpc"
+	"github.com/intelsdi-x/snap/control/strategy"
+	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/cdata"
+	"github.com/intelsdi-x/snap/core/ctypes"
+	"github.com/intelsdi-x/snap/grpc/controlproxy"
+	"github.com/intelsdi-x/snap/internal/common"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestControlGRPCServerRest(t *testing.T) {
```

Comment:
I think the thinking behind this was to have `TestControlGRPCServerX` where X was the component that used the functionality being tested (here is REST). Admittedly not the best name. Open to suggestions for better names.